crates/engine/tree/src/persistence.rs

```diff
@@ -58,44 +67,54 @@ where
         while let Ok(action) = self.incoming.recv() {
             match action {
                 PersistenceAction::RemoveBlocksAbove(new_tip_num, sender) => {
-                    let provider_rw = self.provider.provider_rw()?;
-                    let sf_provider = self.provider.static_file_provider();
-
-                    UnifiedStorageWriter::from(&provider_rw, &sf_provider)
-                        .remove_blocks_above(new_tip_num)?;
-                    UnifiedStorageWriter::commit_unwind(provider_rw, sf_provider)?;
-
+                    let result = self.on_remove_blocks_above(new_tip_num)?;
                     // we ignore the error because the caller may or may not care about the result
-                    let _ = sender.send(());
+                    let _ = sender.send(result);
                 }
                 PersistenceAction::SaveBlocks(blocks, sender) => {
-                    let Some(last_block) = blocks.last() else {
-                        let _ = sender.send(None);
-                        continue
-                    };
-
-                    let last_block_hash = last_block.block().hash();
-
-                    let provider_rw = self.provider.provider_rw()?;
-                    let static_file_provider = self.provider.static_file_provider();
-
-                    UnifiedStorageWriter::from(&provider_rw, &static_file_provider)
-                        .save_blocks(&blocks)?;
-                    UnifiedStorageWriter::commit(provider_rw, static_file_provider)?;
-
+                    let result = self.on_save_blocks(blocks);
                     // we ignore the error because the caller may or may not care about the result
-                    let _ = sender.send(Some(last_block_hash));
+                    let _ = sender.send(result.ok().flatten());
                 }
                 PersistenceAction::PruneBefore(block_num, sender) => {
+                    let start_time = Instant::now();
```

Comment:
let's also move this into prune_before, so that we're consistent with where we track metrics

Reply:
@mattsse done

Reply:
@mattsse what's latency metrics for?